packages/model-viewer/src/features/scene-graph/sampler.ts

```diff
@@ -113,11 +113,11 @@ export class Sampler extends ThreeDOMElement implements SamplerInterface {
     return this[$threeTexture].rotation;
   }
 
-  get scale(): Vector2 {
+  get scale(): Vector2D {
     return this[$threeTexture].repeat;
   }
 
-  get offset(): Vector2|null {
+  get offset(): Vector2D|null {
```

Comment:
I just realized this also needs to change - you're returning a pointer to an internal three.js construct, which means they could then use it to change out internals (in JS every object is pass by reference). You need to use our `toVector2D()` to make a proper copy.